### ISSUE TITLE
Ranged Weaponskills seem to have been hardcoded with exploitable ranges.

### ISSUE THREAD

**Issue by GMYasiinEra** [DarkstarProject/darkstar#3756]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **30161227_1**

**_Server Version_** (type `@revision` in game) **unknown - master clone from last week**

**_Source Branch_** (master/stable) **master**

**_Additional Information_** (Steps to reproduce/Expected behavior) **Tested on ERA and Master DSP**

So considering SQL calls a value of 16 for every ranged ws. I think someone ninja updated ranger ws's in the core. First noticed this after a January update on ERA. reproduced on my test server. Its super OP alot of Linkshells tend to just Ranger Burn everything that have dangerous AoE moves. Id like to see it made a little more interesting and retail like. Wish I understood the core files more to help.

http://imgur.com/a/ofGXY

**Comment by teschnei** [comment]
Ranger burning things with dangerous tp moves is exactly how retail works

**Comment by GMYasiinEra** [comment]
lol yeah i get that but you still had to run to ws and get back out of harms way tactically after mobs skills like  gates etc.right now you can flat out TP and WS at 20+

**Comment by Hozu** [comment]
Is mob AoE range supposed to take monster size into account or no?

**Comment by teschnei** [comment]
You didn't have to move to use a WS. The only ranged WS that needs to be
close is blast shot

On Feb 28, 2017 9:48 AM, "Hozu" <notifications@github.com> wrote:

Is mob AoE range supposed to take monster size into account or no?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/DarkstarProject/darkstar/issues/3756#issuecomment-283095828>,
or mute the thread

**Comment by GMYasiinEra** [comment]
> You didn't have to move to use a WS. The only ranged WS that needs to be
close is blast shot

DB shows Blast Shot as even further away lol `...` 
Wiki shows 5 yalms `...`
http://ffxiclopedia.wikia.com/wiki/Distance

15.8'	
Maximum range for most ranged Weapon Skills (excluding Blast Arrow, Blast Shot).

17.7'	
Maximum range for Detonator and Empyreal Arrow.

17.2'	
Maximum range for Jishnu's Radiance.

Just sayin `...`

Also Coronach falls into the 15.8' category making the database entry of 16 correct `...` 
Like I said before, you actually  TP beyond the range of 20' yalm mob skills and quickly run in to pop off a weapons skill after a mob uses their TP then you gtfo out again... Ranger 101 `...` 
I mean, if you were the one who changed no biggie.. just wanted yas to know its terribly wrong and extremely over powered.

**Comment by teschnei** [comment]
I see, I did not understand your initial comment, then.

Can't do anything until we can confirm whether ranged weaponskills should take monster radius into account, and whether monster aoe moves should also take monster radius into account.

**Comment by GMYasiinEra** [comment]
> and whether monster aoe moves should also take monster radius into account.

@Hozu  nevermind teschnei made me understand your comment.

Yeah dude glad you understand me now.. Basically something in the core is totally by passing the values set in the DB for the maximum range in which you can use various ranged type weapon skills.

**Comment by GMYasiinEra** [comment]
> Can't do anything until we can confirm whether ranged weaponskills should take monster radius into account, and whether monster aoe moves should also take monster radius into account.

I'm all for this remaining the same as there really isn't really another way around it when taking into account melee's, all I'm saying is its definitely hardcoded somewhere right now at 20 in the core and it should be 16 or deleted so the DB can control it.

**Comment by GMYasiinEra** [comment]
Confirmed, mob family size does not increase the AOE range of the mob skills used(tested with Gates of hades with multiple mob sizes, 20 yalms every time. Apply this to ranged ws? Recode rng ws to work off db? Find the hardcode for 20 and set to 16?

**Comment by teschnei** [comment]
I meant on retail :)

**Comment by GMYasiinEra** [comment]
yeah im just saying even with a default radius of 1( a beach bunny, lizzard, anything tiny..) you can still pop off ws from 20 yalms away all i was getting at with the family size was its increaseing it even more to like 21ish

**Comment by Hozu** [comment]
When comparing retail to DSP for distances, remember that this issue also exists: https://github.com/DarkstarProject/darkstar/issues/2904

**Comment by ramenkat** [comment]
I don't know if this will help or not (since I am not a code monkey like ya'll are) but on retail, ranger had a 'sweet spot' for weapon skills, which doesn't seem to apply on DSP servers from my experience.  You can be in melee range and fire off Sidewinder, for example, and it'll be fine, but on retail, you had to be at a distance for the WS to be more accurate/effective.  I wanna say something like.. between 12-15 on the Distance plugin (it's been a while, so I forget for sure)?  I dunno how many 'yalms' that is though.

hope that helps.  if not, feel free to ignore lol

**Comment by Hozu** [comment]
Yeah it's a ranged attack penalty based on distance, and it's different between bow, crossbow, and gun. Problem is there wasn't data for each distance range so it's not implemented. I think there's an outstanding issue open for this.

**Comment by GMYasiinEra** [comment]
tbh i dont care really about any of the other issues.. lol i just want to either see 1 the hardcode removed and allow us to control ws range based on DB again(like it was a few months back) or two change the hardcode from 20' to 16'.  I just watched 6 rangers all standing 20'+ and all unloaded on a dark ixion  and also sarameya and its really sad to see.

**Comment by areckx** [comment]
I'm just starting to browse the source and it looks like a lot of stuff that's hardcoded should be set up dynamically.

A complete overhaul might fix a lot of ooen issues.
